File: app/src/main/java/com/kksionek/photosyncer/view/MenuItemSyncCtrl.java
package com.kksionek.photosyncer.view;

import android.content.Context;
import androidx.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;

import com.kksionek.photosyncer.R;

class MenuItemSyncCtrl {
    private final Context mCtx;
    private final MenuItem mMenuItem;
    private Animation mRotationAnimation = null;
    private ImageView mRefreshImage = null;
    private boolean mAnimating = false;

    MenuItemSyncCtrl(@NonNull Context ctx, @NonNull MenuItem menuItem) {
        mCtx = ctx;
        mMenuItem = menuItem;
    }

    void startAnimation() {
        if (mAnimating)
            return;
        mAnimating = true;
        if (mRefreshImage == null)
            mRefreshImage = (ImageView) LayoutInflater.from(mCtx).inflate(
                    R.layout.refresh_action_view,
                    null);
        if (mRotationAnimation == null)
            mRotationAnimation = AnimationUtils.loadAnimation(mCtx, R.anim.anim_rotate);
        if (mMenuItem.getActionView() == null) {
            mRefreshImage.startAnimation(mRotationAnimation);
            mMenuItem.setActionView(mRefreshImage);
            mMenuItem.setEnabled(false);
        }
    }

    void endAnimation() {
        if (!mAnimating)
            return;
        mAnimating = false;
        if (mMenuItem.getActionView() != null) {
            mMenuItem.getActionView().clearAnimation();
            mMenuItem.setActionView(null);
        }
        mMenuItem.setEnabled(true);
    }
}

File: app/src/main/java/com/kksionek/photosyncer/view/MenuItemSyncCtrl.kt
package com.kksionek.photosyncer.view

import android.content.Context
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.ImageView

import com.kksionek.photosyncer.R

internal class MenuItemSyncCtrl(private val context: Context, private val menuItem: MenuItem) {

    private lateinit var rotationAnimation: Animation
    private lateinit var refreshImage: ImageView
    private var animating = false

    fun startAnimation() {
        if (animating) return
        animating = true
        if (!::refreshImage.isInitialized) {
            refreshImage = LayoutInflater.from(context).inflate(
                R.layout.refresh_action_view,
                null
            ) as ImageView
        }
        if (!::rotationAnimation.isInitialized) {
            rotationAnimation = AnimationUtils.loadAnimation(context, R.anim.anim_rotate)
        }
        if (menuItem.actionView == null) {
            refreshImage.startAnimation(rotationAnimation)
            menuItem.actionView = refreshImage
            menuItem.isEnabled = false
        }
    }

    fun endAnimation() {
        if (!animating) return
        animating = false
        if (menuItem.actionView != null) {
            menuItem.actionView.clearAnimation()
            menuItem.actionView = null
        }
        menuItem.isEnabled = true
    }
}
